docs/developers/reproducible-contract-builds.md

```diff
@@ -0,0 +1,203 @@
+
+h = blake2b(digest_size=32)
+h.update(code)
+codehash_hex = h.hexdigest()
+
+print(codehash_hex)
+```
+
+We can also use the utility `b2sum` to locally compute the _codehash_:
+
+```
+b2sum -l 256 adder.wasm
+```
+
+Now, it's easy to verify the bytecode equality of two given builds of a contract using solely the _codehash_ property.
+
+## Supporting reproducible builds
+
+As of September 2022, the recommended approach to support reproducible builds for your smart contract is to perform the **deployment build** (as opposed to regular development builds) in a Docker container based on a [publicly-available, tagged Docker image](https://hub.docker.com/r/elrondnetwork/elrond-sdk-erdpy-rust/tags), that includes tagged, explicit versions of the build tools (_erdpy_, _Rust_, _wasm-opt_ etc.). In addition, the current directory when running the build command should be: `/home/elrond`. 
+
+The above constraints are recommended in order to counteract eventual pieces of non-determinism related to `cargo`'s (essential component of the Rust toolchain) sensibility on the environment.
+
+:::important
+If the code source of your smart contract is hosted on GitHub, then it's a good practice to define a GitHub Workflow similar to [this one](https://github.com/ElrondNetwork/reproducible-contract-build-example/blob/main/.github/workflows/release.yml), which performs the deployment (production-ready) build within the _release_ procedure.
+:::
+
+### Choose an image tag
+
+The first step for supporting reproducible builds is to decide on a specific Docker image tag to use. Check the **frozen** tags listed at [elrondnetwork/elrond-sdk-erdpy-rust](https://hub.docker.com/r/elrondnetwork/elrond-sdk-erdpy-rust/tags), and inspect their labels - especially the label `rust`:
+
+```
+LABEL rust=nightly-2022-03-01
+```
+
+The chosen, _frozen_ image tag should accompany the versioned source code (e.g. via _release notes_), in order to inform others on how to reproduce a specific build (of a specific source code version). In this context, _frozen_ image tag refers to a Docker image tag that will never get any updates after its initial publishing.
+
+:::tip
+If unsure about which one to choose - pick the tag with the **largest index number**. 
+:::
+
+:::warning
+Never pick the tag called `latest` for production-ready builds.
+:::
+
+### Regular builds vs. _deployment_ build
+
+Usually, while developing, testing and integrating a smart contract, one would simply run locally the command `erdpy contract build`, which generates the WASM bytecode. 
+
+However, for production-ready deployments, in order to achieve reproducible builds, make sure to only deploy contracts built as described in the next section, using the chosen Docker image (tag).
+
+## (Re)producing a build
+
+In this section, you'll learn how to run a reproducible build, or, to put it differently, how to reproduce a previous build (made by you or by someone else in the past), on the local machine.
```

Comment:
This section isn't just for reproducible builds, but also for building contracts without a pre-defined environment. Therefore, might rename the section to `## Building a contract via Docker (reproducible builds)`
You might also explain a bit the advantage that no environment is needed

Reply:
Fixed.